### ISSUE TITLE
Feature Request: Support for JupyterLab

### ISSUE THREAD

**Issue by DJuboor** [harvard/cloudJHub#9]

**Comment by alexraju91** [comment]
Just found out how to do this. 
Add the following lines to the make_worker_ami method in `launch.py` script.
```
sudo("pip3 --quiet install ipython jupyter jupyterlab ipykernel nbgrader")
sudo("jupyter serverextension enable --py jupyterlab --sys-prefix")
```
`jupyterhub_config.py`

```
c.Spawner.default_url = '/lab'
c.Spawner.cmd = ['jupyterhub-singleuser']
```

**Comment by farassadek** [comment]
Thank you Alex , this should make it work. 
If you faced any issue please let me know

**Comment by h4gen** [comment]
Indeed the solution proposed by @alexforever86 provides a jupyterlab instance, but in my case the hub bar is vanished. Do you also encounter this problem?

**Comment by farassadek** [comment]
Hi 
I used the lab with the older version of Jupyterhub and was working , It might be the new version ? I did not test it currently. 
@alexforever86  did you test your solution with the current version of Jupyterhub ?

**Comment by h4gen** [comment]
I also tried this https://github.com/jupyterhub/jupyterlab-hub extension but it did not really get me further :/ In fact I got an 403 error when entering the lab. Maybe it has something todo with the `page_config.json` part. I have to admit I do not really understand what it is about or if this condition is fulfilled in your setup.

**Comment by farassadek** [comment]
Did you install jupyterlab-hub on the AMI that used to create the user instances ?

**Comment by h4gen** [comment]
Yes, exactly! I added these lines and the installation works without problems on the AMI:
`sudo("pip3 --quiet install ipython jupyter jupyterlab ipykernel nbgrader")`
`sudo("curl -sL https://deb.nodesource.com/setup_10.x | sudo -E bash -") `
`sudo("apt-get install -qq -y nodejs")`
`sudo("jupyter labextension install @jupyterlab/hub-extension")`

**Comment by farassadek** [comment]
I cannot tell for definite what is going wrong since there are many parts need to come together. Therefore I will create a cluster myself and check jupyterlab with the latest version of jupyterhub. 
Is this urgent or it can wait (I am a bit busy those days preparing the lectures for my devops class that starts in 10 days)? If it is urgent , can I have access to  your cluster to check what is going on ?

**Comment by h4gen** [comment]
No its really not that urgent! I will try a bit more and also focus on some other problems in the meantime. Thank you a lot!

**Comment by farassadek** [comment]
I will leave this issue open until we find a resolution. If you are able to make the lab works please share it so I can added to the documentation. 
Thanks and sorry if I was not of a help here

**Comment by alexraju91** [comment]
@h4gen I tried this with old version. Which `cmd` parameter did you use with this? 

``` c.Spawner.cmd = ['jupyter-labhub']```

or 

``` c.Spawner.cmd = ['jupyterhub-singleuser']```

**Comment by h4gen** [comment]
I tried both!
Your version leads to normal behaviour but also without the hub bar.
The 
`c.Spawner.cmd = ['jupyter-labhub']`
configuration leads to no hub bar and no usability as a few dialugue windows appear:

Trying to start notebook:
![image](https://user-images.githubusercontent.com/12556763/44625781-7ce94280-a911-11e8-8d47-cd77e416cfee.png)

![image](https://user-images.githubusercontent.com/12556763/44625792-99857a80-a911-11e8-99a5-71fc4013d0a4.png)

Open a file from the file menue:

![image](https://user-images.githubusercontent.com/12556763/44625796-b91ca300-a911-11e8-842d-c6ea0142539d.png)


I guess it has to do with the `page_config.json` mentioned on the labhub page. I tried to inject

`{
    "hub_prefix": "/jupyter"
}`

but I think this is the wrong path and I don't really understand what would be correct in our case.
